src/lib.rs

```diff
@@ -191,41 +191,59 @@ pub mod pallet {
 				Error::<T>::InvalidClaimSignature
 			);
 
-			// We ensure the relay chain id wast not yet associated to avoid multi-claiming
-			ensure!(
-				ClaimedRelayChainIds::<T>::get(&relay_account).is_none(),
-				Error::<T>::AlreadyAssociated
-			);
+			let mut reward_info =
+				// If an association exists, we just need to insert it in the new reward address
+				if let Some(associated) = ClaimedRelayChainIds::<T>::get(&relay_account) {
+					// Get reward info
+					let reward_info = AccountsPayable::<T>::get(&associated)
+						.ok_or(Error::<T>::NoAssociatedClaim)?;
+					// Remove from associated
+					AccountsPayable::<T>::remove(&associated);
```

Comment:
Why are removing this? I see that we `insert()` later (line 243), does that make this removal unneeded?

Reply:
Nevermind, I see -- that's the whole point of this change. We remove it, take the opportunity to pay out, and then reinsert it with the updated address

Reply:
Actually, we don't pay out in this case. Should we, since we do so in the other `else {` branch?

Reply:
No, we only pay out if it's the first time this address is being associated. So this function takes two branches

- Either a parachain address was not provided when contributing to the crowdloan, in which case we will insert the relay address into the UnassociatedContributions storage. When someone wants to associate a parachain address to an unassociated relay chain address,  then we pay out the first 30% of the total rewards (we cannot do it before since we dont have the para address to send the rewards)

- If this function is used not to associate an unassociated relay address but to change the reward address associated already, then we just update the mapping (we remove the previous associated reward address and insert the reward info in the new one)